pkg/models/v1/contacts.go

```diff
@@ -311,19 +311,8 @@ func GetSentEmailCount(contact *pb.Contact, reason string, timeWindowDays int) (
 		return 0, err
 	}
 
-	for _, value := range emailLog {
-		strTimestamp := value.Timestamp
-		timestamp, err := time.Parse(time.RFC3339, strTimestamp)
-		if err != nil {
-			return 0, fmt.Errorf("error parsing timestamp: %v", err)
-		}
-
-		matchedReason := reason == value.Reason
-		withinTimeWindow := timestamp.After(time.Now().AddDate(0, 0, -timeWindowDays))
-
-		if matchedReason && withinTimeWindow {
-			sent++
-		}
+	if sent, err = countSentEmails(emailLog, reason, timeWindowDays); err != nil {
+		return 0, err
```

Comment:
This is great, I'm happy we're getting rid of the duplicated code. To take this a step farther, we could probably also remove the `GetSentEmailCount` and `GetSentAdminEmailCount` in favor of the new helper. This means that we have to fetch the email log wherever those were being called but I think it will reduce the bloat in the models package a bit more and prevent us from having to pass through the `reason` and `timeWindowDays`. What do you think?